### PR TITLE
[test] Fixup some tests for older OSes and other inconsequential changes

### PR DESCRIPTION
**Pull request by milseman**

### CODE REVIEW

**Comment by milseman** [comment]
@swift-ci please test

**Comment by milseman** [comment]
@swift-ci please smoke test and merge

File: test/stdlib/Character.swift [hunk 1]
```diff
@@ -105,18 +105,17 @@ testCharacters += [
...
-  "\u{00a9}\u{0300}\u{0300}\u{0300}\u{0300}\u{0300}", // UTF-8: 12 bytes
-  "\u{00a9}\u{0300}\u{0300}\u{0300}\u{0300}\u{0300}\u{0300}", // UTF-8: 14 bytes
-  "\u{00a9}\u{0300}\u{0300}\u{0300}\u{0300}\u{0300}\u{0300}\u{0300}", // UTF-8: 16 bytes
-
-  "👩🏽‍💼", // UTF-8: 15 bytes
-  "👩‍👩‍👦‍👦", // UTF-8: 25 bytes
-]
-#endif
+// Only run it on recent enough versions of ICU
+if #available(iOS 11.0, macOS 10.13, *) {
```
Comment (jrose-apple) [inline test/stdlib/Character.swift]: This should guard against the equivalent tvOS and watchOS versions too.

**Comment by milseman** [comment]
@swift-ci please smoke test

**Comment by milseman** [comment]
@swift-ci please smoke test linux platform

**Comment by milseman** [comment]
Linux failure unrelated. Merging to unblock bots.

**Pull request merged**